src/coreclr/gc/gc.cpp

```diff
@@ -24965,15 +24972,8 @@ void gc_heap::check_heap_count ()
 {
     dynamic_heap_count_data.new_n_heaps = n_heaps;
 
-    if (GCConfig::GetHeapCount() != 0)
-    {
-        // don't change the heap count if we see an environment variable setting it explicitly
-        return;
-    }
-
-    if (GCConfig::GetGCDynamicAdaptationMode() == 0)
+    if (dynamic_adaptation_mode != dynamic_adaptation_to_application_sizes)
```

Comment:
should this check the default instead of `!= dynamic_adaptation_to_application_sizes` so when we add additional modes this doesnt need to change?

Reply:
the reason why I intentionally only compared with this mode is because I don't know if the other modes would require this work (that part isn't decided yet at all). so I'm only comparing with the mode we implemented. does that make sense?

Reply:
ok sure.